pgml-dashboard/src/components/chatbot/chatbot_controller.js

```diff
@@ -0,0 +1,184 @@
+  };
+}
+
+// From: https://github.com/jackmoore/autosize, wrapped in an anonymous function and then minified
+const autosize = function() { let e = new Map; function t(t) { let o = e.get(t); o && o.destroy() } function o(t) { let o = e.get(t); o && o.update() } let r = null; return "undefined" == typeof window ? ((r = e => e).destroy = e => e, r.update = e => e) : ((r = (t, o) => (t && Array.prototype.forEach.call(t.length ? t : [t], t => (function t(o) { if (!o || !o.nodeName || "TEXTAREA" !== o.nodeName || e.has(o)) return; let r = null, l = window.getComputedStyle(o); function n({ restoreTextAlign: e = null, testForHeightReduction: t = !0 }) { let i = l.overflowY; if (0 === o.scrollHeight) return; "vertical" === l.resize ? o.style.resize = "none" : "both" === l.resize && (o.style.resize = "horizontal"); let s; t && (s = function e(t) { let o = []; for (; t && t.parentNode && t.parentNode instanceof Element;)t.parentNode.scrollTop && o.push([t.parentNode, t.parentNode.scrollTop]), t = t.parentNode; return () => o.forEach(([e, t]) => { e.style.scrollBehavior = "auto", e.scrollTop = t, e.style.scrollBehavior = null }) }(o), o.style.height = ""); let d; if (d = "content-box" === l.boxSizing ? o.scrollHeight - (parseFloat(l.paddingTop) + parseFloat(l.paddingBottom)) : o.scrollHeight + parseFloat(l.borderTopWidth) + parseFloat(l.borderBottomWidth), "none" !== l.maxHeight && d > parseFloat(l.maxHeight) ? ("hidden" === l.overflowY && (o.style.overflow = "scroll"), d = parseFloat(l.maxHeight)) : "hidden" !== l.overflowY && (o.style.overflow = "hidden"), o.style.height = d + "px", e && (o.style.textAlign = e), s && s(), r !== d && (o.dispatchEvent(new Event("autosize:resized", { bubbles: !0 })), r = d), i !== l.overflow && !e) { let a = l.textAlign; "hidden" === l.overflow && (o.style.textAlign = "start" === a ? "end" : "start"), n({ restoreTextAlign: a, testForHeightReduction: !0 }) } } function i() { n({ testForHeightReduction: !0, restoreTextAlign: null }) } let s, d = (s = o.value, () => { n({ testForHeightReduction: "" === s || !o.value.startsWith(s), restoreTextAlign: null }), s = o.value }), a = (t => { o.removeEventListener("autosize:destroy", a), o.removeEventListener("autosize:update", i), o.removeEventListener("input", d), window.removeEventListener("resize", i), Object.keys(t).forEach(e => o.style[e] = t[e]), e.delete(o) }).bind(o, { height: o.style.height, resize: o.style.resize, textAlign: o.style.textAlign, overflowY: o.style.overflowY, overflowX: o.style.overflowX, wordWrap: o.style.wordWrap }); o.addEventListener("autosize:destroy", a), o.addEventListener("autosize:update", i), o.addEventListener("input", d), window.addEventListener("resize", i), o.style.overflowX = "hidden", o.style.wordWrap = "break-word", e.set(o, { destroy: a, update: i }), i() })(t, o)), t)).destroy = e => (e && Array.prototype.forEach.call(e.length ? e : [e], t), e), r.update = e => (e && Array.prototype.forEach.call(e.length ? e : [e], o), e)), r }();
```

Comment:
You'd want to import this here as a module, if possible: https://github.com/postgresml/postgresml/blob/master/pgml-dashboard/templates/layout/head.html#L39

So other components can reuse this too.